packages/dd-trace/src/span_pointers.js

```diff
@@ -0,0 +1,27 @@
+const crypto = require('crypto')
+
+const S3_PTR_KIND = 'aws.s3.object'
+
+const SPAN_POINTER_DIRECTION = Object.freeze({
+  UPSTREAM: 'u',
+  DOWNSTREAM: 'd'
+})
+
+/**
+ * Generates a unique hash from an array of strings by joining them with | before hashing.
+ * Used to uniquely identify AWS requests for span pointers.
+ * Expects S3 ETag to already have quotes removed!
+ * @param {string[]} components - Array of strings to hash
+ * @returns {string} A 32-character hash uniquely identifying the components
+ */
+function generatePointerHash (components) {
```

Comment:
NIT: rather then adding a new file, would it make more sense for this function to be in:

packages/datadog-plugin-aws-sdk/src/base.js
Or
packages/dd-trace/src/util.js

Reply:
We can't do base.js because we need to access it from dd-trace in https://github.com/DataDog/datadog-lambda-js, but we can move it to util.js!